core/src/main/java/tc/oc/pgm/flag/state/Respawning.java

```diff
@@ -52,15 +57,30 @@ public void enterState() {
 
     if (!Duration.ZERO.equals(respawnTime)) {
       // Respawn is delayed
-      this.flag
-          .getMatch()
-          .sendMessage(
-              new PersonalizedTranslatable(
-                  "flag.willRespawn",
-                  this.flag.getComponentName(),
-                  new PersonalizedText(
-                      PeriodFormats.briefNaturalApproximate(respawnTime),
-                      net.md_5.bungee.api.ChatColor.AQUA)));
+
+      String postName = this.respawnToPost.getPostName();
+      Component timeComponent =
+          new PersonalizedText(
+              PeriodFormats.briefNaturalApproximate(respawnTime),
+              net.md_5.bungee.api.ChatColor.AQUA);
+      String timeStr = timeComponent.render().toLegacyText();
+
+      if (postName != null) {
+        this.flag
+            .getMatch()
+            .sendMessage(
+                TranslatableComponent.of("flag.willRespawn.named")
+                    .args(
+                        this.flag.getTextName(),
+                        TextComponent.of(postName, TextColor.AQUA),
+                        TextComponent.of(timeStr)));
+      } else {
+        this.flag
+            .getMatch()
+            .sendMessage(
+                TranslatableComponent.of("flag.willRespawn")
+                    .args(this.flag.getTextName(), TextComponent.of(timeStr)));
+      }
```

Comment:
You can just do `TranslatableComponent.of("key", arg1, arg2, arg3)` instead of `TranslatableComponent.of("key").args(arg1, arg2, arg3)`

Reply:
Got it. Changed it 👍